vmdb/spec/models/ems_event/parsers/openstack_infra_spec.rb

```diff
@@ -0,0 +1,123 @@
+require "spec_helper"
+
+describe EmsEvent::Parsers::OpenstackInfra do
+
+  context ".event_to_hash" do
+    it "with a compute.instance.create.end event" do
+      event = YAML.load_file(File.join(evp_data_dir, 'compute_instance_create_end.yml'))
+      data = described_class.event_to_hash(event, 123)
+
+      expected_attributes = common_attributes(event).merge(
+        :event_type   => "compute.instance.create.end",
+        :chain_id     => "r-otxomvqw",
+        :timestamp    => "2015-05-12 07:24:39.462895",
+        :host_ems_ref => "cdab9a8d-d653-4dee-81f9-173f9a22bd2e",
+        :message      => "Success"
+      )
+
+      data.should have_attributes(expected_attributes)
+
+      data[:full_data].should    be_instance_of Hash
+      data[:host_ems_ref].should be_instance_of String
```

Comment:
This last piece is redundant because you are testing the attribute directly...not a big deal though.